### PR TITLE
Remove `ParsePromise` in favor of ES2015 `Promise`

### PR DESCRIPTION
**Pull request by jaalzateolaya**
`Parse.Promise` gets removed in
parse-community/Parse-SDK-JS@c344fd4779a81c621ed4ea14853060079445ecb8

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: parse-community/Parse-SDK-JS@c344fd4779a81c621ed4ea14853060079445ecb8
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by alexander-alzate** [comment]
The test is failing because a type definition which doesn't really need to be updated since `parse-mockdb` package is still using an old version of Parse.

How am I supposed to fix that?

**Comment by SimonSchick** [comment]
Move the old typings into the `v1` folder and add redirect, see `hapi` for examples.

**Comment by alexander-alzate** [comment]
The test worked locally, what's wrong now?

I'm not really sure about how that versioning system works, I guessed it by comparing `hapi` code as pointed by @SimonSchick .

**Comment by alexander-alzate** [comment]
I'm really stuck from here.

File: types/parse-mockdb/tsconfig.json [hunk 1]
```diff
@@ -13,6 +13,11 @@
         "typeRoots": [
             "../"
         ],
+        "paths": {
```
Comment (SimonSchick) [inline types/parse-mockdb/tsconfig.json]: This is only needed in the v1 folder.

**Review by SimonSchick — COMMENTED** [review]
Please consider doing `If you are making substantial changes, consider adding a`, this package has most rules disabled which could hide some issues.

**Pull request closed without merging**